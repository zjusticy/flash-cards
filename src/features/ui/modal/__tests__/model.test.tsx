import React from 'react';
// import renderer from "react-test-renderer";
// import configureStore from 'redux-mock-store'
// import { shallow } from 'enzyme';
// import toJson from 'enzyme-to-json';
import { afterEach, describe, it, expect } from 'vitest';

import { render, cleanup } from '@testing-library/react';

import Modal from '../modal';

afterEach(cleanup);

describe('Button', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Modal show modalClosed={() => {}}>
        Facebook
      </Modal>
    );
    // expect(getByRole("button").classList.contains("Big")).toBe(true);
    // expect(getByRole("button").textContent).toEqual("Facebook");

    expect(asFragment()).toMatchSnapshot();
  });
});
