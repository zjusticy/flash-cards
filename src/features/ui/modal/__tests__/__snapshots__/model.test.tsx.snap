// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-label="close Modal"
    class="w-full h-full fixed z-[100] left-0 top-0 bg-black bg-opacity-50"
    role="button"
    tabindex="0"
  />
  <div
    class="modal"
    style="transform: translateY(0); opacity: 1;"
  >
    Facebook
  </div>
</DocumentFragment>
`;
